archeryutils/handicaps/handicap_scheme.py

```diff
@@ -0,0 +1,764 @@
+
+        Returns
+        -------
+        handicap : int or float
+            Handicap for maximum score. Has type int if int_prec is True, else float.
+
+        Warns
+        ------
+        UserWarning
+            If called with int_prec=False as value limited by the numerical scheme delta.
+
+        """
+        max_score = rnd.max_score()
+
+        if self.desc_scale:
+            handicap = -75.0
+            delta_hc = 0.01
+        else:
+            handicap = 175.0
+            delta_hc = -0.01
+
+        # Set rounding limit
+        if self.name in ("AGB"):
+            round_lim = 1.0
+        else:
+            round_lim = 0.5
+
+        s_max = self.score_for_round(rnd, handicap, arw_d, rounded_score=False)
+        # Work down to where we would round or ceil to max score
+        while s_max > max_score - round_lim:
+            handicap = handicap + delta_hc
+            s_max = self.score_for_round(rnd, handicap, arw_d, rounded_score=False)
+        handicap = handicap - delta_hc  # Undo final iteration that overshoots
+        if int_prec:
+            if self.desc_scale:
+                handicap = np.floor(handicap)
+            else:
+                handicap = np.ceil(handicap)
+        else:
+            warnings.warn(
+                "Handicap requested for maximum score without integer precision.\n"
+                "Value returned will be first handicap that achieves this score.\n"
+                "This could cause issues if you are not working in integers.",
+                UserWarning,
+            )
+        return handicap
+
+    def _rootfind_score_handicap(
+        self,
+        score: float,
+        rnd: rounds.Round,
+        arw_d: Optional[float] = None,
+    ) -> float:
+        """Get handicap for general score on a round through rootfinding algorithm.
+
+        Parameters
+        ----------
+        score : float
+            score to get handicap for
+        rnd : rounds.Round
+            round being used
+        arw_d : float or None, default=None
+            user-specified arrow diameter in [metres]
+
+        Returns
+        -------
+        handicap : float
+            appropriate accurate handicap for this score
+
+        References
+        ----------
+        Brent's Method for Root Finding in Scipy:
+
+        - https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.brentq.html
+        - https://github.com/scipy/scipy/blob/dde39b7/scipy/optimize/Zeros/brentq.c
+
+        """
+        # The rootfinding algorithm here raises pylint errors for
+        # too many statements (64/50), branches (17/12), and variables(23/15).
+        # However, it is a single enclosed algorithm => disable
+        # pylint: disable=too-many-locals
+        # pylint: disable=too-many-branches
+        # pylint: disable=too-many-statements
+
+        if self.desc_scale:
+            x_init = [-75.0, 300.0]
+        else:
+            x_init = [-250.0, 175.0]
```

Comment:
Same point about coupling desc_scale to specifics of handicap system search parameters